internal/server/query_test.go

```diff
@@ -0,0 +1,125 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/carlmjohnson/be"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgproto3"
+
+	"github.com/mattermost/perseus/config"
+	"github.com/mattermost/perseus/internal/pgmock"
+)
+
+func TestHandleQuery(t *testing.T) {
+	ln, err := net.Listen("tcp", "127.0.0.1:")
+	be.NilErr(t, err)
+	serverErrChan := make(chan error, 1)
+	defer ln.Close()
+
+	var conn net.Conn
+	defer func() { conn.Close() }()
```

Comment:
What's the difference between `defer conn.Close()`?

Reply:
The difference is that, the value of `conn` gets immediately bound. And initially that value is null. Therefore when the defer calls, it throws a panic. So this is a way to delay resolution of the variable until actually when the function is called.

Reply:
TIL.